### ISSUE TITLE
[CP] [Impeller] Fix contiguous clip restoration

### ISSUE THREAD

**Issue by zanderso** [flutter/flutter#118800]
### issue_link

N/A

### Commit Hash

https://github.com/flutter/engine/commit/0d79533876c7e10ad28bca48d65ba4419f444dee

### Target

stable

### pr_link

https://github.com/flutter/engine/pull/39002

### Impacted Users

Impeller users

### Impact Description

In correct rendering in the presence of a save layer containing multiple clips.

### Workaround

There is no workaround

### Risk

low

### Test Coverage

yes

### Validation Steps

Assets from an internal asset mentioned in chat render correctly.

**Comment by zanderso** [comment]
Applying this is likely blocked by https://github.com/flutter/flutter/issues/118805

**Comment by XilaiZhang** [comment]
pr landed in release branch successfully

**Comment by bdero** [comment]
@zanderso I just noticed that this fix trades off for another issue that could result in excess draw culling: https://github.com/flutter/engine/pull/39023

**Comment by CaseyHillers** [comment]
This was shipped in the 3.7.0-1.5 beta
